graph-proxy/src/graphql/workflows.rs

```diff
@@ -185,6 +189,114 @@ impl TryFrom<IoArgoprojWorkflowV1alpha1WorkflowStatus> for WorkflowErroredStatus
     }
 }
 
+#[allow(clippy::missing_docs_in_private_items)]
+#[derive(Debug, Enum, PartialEq, Eq, Clone, Copy)]
+enum TaskStatus {
+    Pending,
+    Running,
+    Succeeded,
+    Skipped,
+    Failed,
+    Error,
+    Omitted,
+}
+
+impl TryFrom<String> for TaskStatus {
+    type Error = WorkflowParsingError;
+
+    fn try_from(status: String) -> Result<Self, <Self as TryFrom<String>>::Error> {
+        match status.as_str() {
+            "Pending" => Ok(TaskStatus::Pending),
+            "Running" => Ok(TaskStatus::Running),
+            "Succeeded" => Ok(TaskStatus::Succeeded),
+            "Skipped" => Ok(TaskStatus::Skipped),
+            "Failed" => Ok(TaskStatus::Failed),
+            "Error" => Ok(TaskStatus::Error),
+            "Omitted" => Ok(TaskStatus::Omitted),
+            _ => Err(WorkflowParsingError::UnrecognisedPhase),
+        }
+    }
+}
+
+/// A Task created by a workflow
+#[derive(Debug, SimpleObject)]
+struct Task {
+    /// Unique name of the task
+    id: String,
+    /// Display name of the task
+    name: String,
+    /// Current status of a task
+    status: TaskStatus,
+    /// Parent of a task
+    depends: Vec<String>,
+}
+
+/// A wrapper for list of tasks
+struct TaskMap(HashMap<String, IoArgoprojWorkflowV1alpha1NodeStatus>);
```

Comment:
Do we want a new type here or just a type alias?

Reply:
I have custom methods, so we need struct? 🤔

Reply:
Yeah, seems reasonable. Other option would be to just inline the dependency resolution into the `TryFrom`

Reply:
But we also have another custom method `generate_relationship_map`?  is it gonna make much difference? is't this easier to work with?